src/Blocks/BlockButton.h

```diff
@@ -145,14 +145,19 @@ class cBlockButtonHandler :
 
 	virtual bool CanBeAt(cChunkInterface & a_ChunkInterface, const Vector3i a_RelPos, const cChunk & a_Chunk) override
 	{
-		auto Meta = a_Chunk.GetMeta(a_RelPos);
-		auto SupportRelPos = AddFaceDirection(a_RelPos, BlockMetaDataToBlockFace(Meta), true);
+		NIBBLETYPE Meta = a_Chunk.GetMeta(a_RelPos);
+		Vector3i SupportRelPos = AddFaceDirection(a_RelPos, BlockMetaDataToBlockFace(Meta), true);
```

Comment:
Why are you removing `auto`?

Reply:
Personally getting into the code for the first time this auto confused me for a bit. Why use auto for trivial types? especially when forwarded to other functions as parameters.
It's also inconsistent with the other CanBeAt functions.

Reply:
Auto has advantages even with trivial types, it prevents unintentional casts when the type on the LHS is not the same as the RHS type.

Your supposed confusion will only be present on literally the first read-through of the code, just use auto everywhere it works.